hera_cal/lstbin.py

```diff
@@ -1000,22 +1002,24 @@ def sigma_clip(array, flags=None, sigma=4.0, axis=0, min_N=4):
     # create empty clip_flags array
     clip_flags = np.zeros_like(array, np.bool)
 
+    # make copy of data array as to not replace original array
+    array_ = array.copy()
```

Comment:
We have to make a copy as to not change input array.

Reply:
let's re-think if this is really necessary, as stated above. memory performance for the LST binner is generally crucial, so saving memory wherever possible is usually helpful

Reply:
I think this is necessary if we still want `flags` as an input, as we don't want to change the original input array, as this would impact sigma clipping for the Im component if Re is done first

Since the `flags` arg is not used in the routine (because `d` already has nan entries for flagged points), we could get rid of the `flags` arg and the `array` here would not have to be copied